pkg/virt-launcher/virtwrap/api/converter_test.go

```diff
@@ -2202,6 +2202,135 @@ var _ = Describe("Converter", func() {
 		})
 	})
 
+	Context("Kernel Boot", func() {
+		var vmi *v1.VirtualMachineInstance
+		var c *ConverterContext
+
+		BeforeEach(func() {
+			vmi = &v1.VirtualMachineInstance{}
+
+			v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+
+			c = &ConverterContext{
+				VirtualMachine: vmi,
+				UseEmulation:   true,
+			}
+		})
+
+		Context("when kernel boot is not set", func() {
+			It("should configure normal boot", func() {
+				vmi.Spec.Domain.Firmware = &v1.Firmware{}
+				domainSpec := vmiToDomainXMLToDomainSpec(vmi, c)
+				Expect(domainSpec.OS.Kernel).To(BeNil())
+				Expect(domainSpec.OS.Initrd).To(BeNil())
+				Expect(domainSpec.OS.Cmdline).To(BeNil())
+			})
+		})
+
+		Context("when kernel boot is set and kernel is provided", func() {
```

Comment:
Maybe you can refactor this by using Table driven tests, this could greatly reduce redundancy, have a look at that section in [this document](https://onsi.github.io/ginkgo/). Get back to me if you have questions.

Reply:
It's located under Extensions on the page mentioned.